### PR TITLE
Customizer

### PR DESCRIPTION
**Pull request by harsh242**
**What does this PR do?**
Implements Customizer UI

**Description of Task to be completed?**
- [ ] Load Cryptobot
- [ ] Customizer Functionality

**How should this be manually tested?**
`tezos/customizebot`

**Any background context you want to provide?**
@bhaskarSingh I am not able to load cryptobot😅 and maybe we can connect tomorrow and discuss. Cryptobot GLB file is available in the drive if you wish to test.

**Screenshots(optional)**
image

### CODE REVIEW

File: src/pages/tezos/customizebot.js [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+            );
+          })}
+      </group>
+    </>
+  );
+};
+
+const Bot = ({ headCount, legCount, bodyCount }) => {
+  const group = useRef();
+  const { scene } = useGLTF('cryptobot.glb');
```
Comment (bhaskarSingh) [inline src/pages/tezos/customizebot.js:76]: @harsh242  Where is the cryptobot.glb file located?

Comment (harsh242) [reply src/pages/tezos/customizebot.js:76]: Locally I have it inside the public folder.

**Comment by bhaskarSingh** [comment]
Try putting it in the static folder that should do the trick. Public folder
is created at build time. You can refer https://youtu.be/1rP3nNY2hTo to get
a sense of how it works in gatsby.

**Comment by harsh242** [comment]
@bhaskarSingh Try swapping the hands.
Is this the same issue?
https://github.com/pmndrs/react-three-fiber/discussions/881

**Comment by bhaskarSingh** [comment]
> @bhaskarSingh Try swapping the hands.
Is this the same issue?
pmndrs/react-three-fiber#881

Probably, try to compare the model structure with the custom model example I worked on(https://github.com/bhaskarSingh/bot_swappable_interface). Check if there's any discrepancy or not b/w the two.

**Comment by harsh242** [comment]
image
Now, cryptobots load perfectly, it was breaking earlier because we weren't adding the rotation of the mesh.

**Comment by bhaskarSingh** [comment]
@manangouhari  Why is suspense needed? It's loading just fine currently &  it was loading just fine till 3066588 commit also.

**Comment by manangouhari** [comment]
> @manangouhari  Why is suspense needed? It's loading just fine currently &  it was loading just fine till 3066588 commit also.

Suspense was needed because we needed to lift-up state to page component, as `useGLTF` was required to export the bot when user claims it. 

What I ended up doing is, repeat state in both `Bot` and `BotClaimBtn`.

The issue now is with Auth. That's broken.

**Comment by bhaskarSingh** [comment]
Got it. Let me make it easier for you. I'll take lead here in creating the customizer + finishing the claim bot functionality. 

Can you take lead for whatever the next critical constraint is?

**Comment by manangouhari** [comment]
> Got it. Let me make it easier for you. I'll take lead here in creating the customizer + finishing the claim bot functionality. 
> 
> Can you take lead for whatever the next critical constraint is? 

No no, it's okay. I've got the Claim Bot flow figured out. Please take lead on the customizer. I'll complete the flow.

**Comment by bhaskarSingh** [comment]
> The issue now is with Auth. That's broken.

It's occurring after the claim bot logic commit. So, probably that's where the issue.

>No no, it's okay. I've got the Claim Bot flow figured out.

It's okay, I will have to work around it anyways since auth is broken.

Take lead for the next-critical which would be finalizing the academy(A lot of work is left here) + landing page.

**Comment by manangouhari** [comment]
> > The issue now is with Auth. That's broken.
> 
> It's occurring after the claim bot logic commit. So, probably that's where the issue.
> 
> >No no, it's okay. I've got the Claim Bot flow figured out.
> 
> It's okay, I will have to work around it anyways since auth is broken.
> 
> Take lead for the next-critical which would be finalizing the academy(A lot of work is left here) + landing page. ✅ if you concur.

Okay, makes sense.

**Pull request closed without merging**